code/modules/mob/living/simple_animal/bot/cleanbot.dm

```diff
@@ -116,6 +148,21 @@
 			bot_patrol()
 
 	if(target)
+		if(QDELETED(target) || !isturf(target.loc))
+			target = null
+			mode = BOT_IDLE
+			return
```

Comment:
I feel like this should be done way earlier. So that the bot can select a new target in the same tick.